r/src/RecordBatch.cpp

```diff
@@ -63,17 +63,26 @@ List RecordBatch__to_dataframe(const std::shared_ptr<arrow::RecordBatch>& batch)
 }
 
 // [[Rcpp::export]]
-std::shared_ptr<arrow::RecordBatch> read_record_batch_(std::string path) {
-  std::shared_ptr<arrow::io::ReadableFile> stream;
+std::shared_ptr<arrow::RecordBatch> read_record_batch_RandomAccessFile(
```

Comment:
We need to revisit these APIs -- read_record_batch_RandomAccessFile and read_record_batch_BufferReader do different things. https://issues.apache.org/jira/browse/ARROW-3498